x-pack/plugins/lens/public/shared_components/legend/legend_settings_popover.tsx

```diff
@@ -209,7 +238,27 @@ export function LegendSettingsPopover<T = XYLegendValue>({
   legendSize,
   onLegendSizeChange,
   showAutoLegendSizeOption,
-}: LegendSettingsPopoverProps<T>) {
+  titlePlaceholder,
+}: LegendSettingsPopoverProps<LegendStats>) {
+  const isLegendNotHidden = mode !== 'hide';
+
+  const canShowStatisticsConfigs = isLegendNotHidden && allowedLegendStats.length > 1;
+
+  const canShowShowValueConfig =
+    isLegendNotHidden &&
+    allowedLegendStats.length === 1 &&
+    (allowedLegendStats[0].value === LegendValue.CurrentAndLastValue ||
+      allowedLegendStats[0].value === LegendValue.Value);
+
+  const isCurrentAndLastValueOnlySelected =
+    legendStats.length === 1 && legendStats[0] === LegendValue.CurrentAndLastValue;
+
+  const canShowLegendTitleChangeConfig =
+    canShowStatisticsConfigs &&
+    !isCurrentAndLastValueOnlySelected &&
+    legendStats.length > 0 &&
+    !!onLegendTitleChange;
```

Comment:
nit: just trying to make sense of the code here, I've tried to rewrite it

```suggestion
  const isSingleStatsOnly = isLegendNotHidden && allowedLegendStats.length === 1;
  const canShowStatisticsConfigs = isLegendNotHidden && allowedLegendStats.length > 1;

  const canShowShowValueConfig =
    isSingleStatsOnly && [LegendValue.CurrentAndLastValue, LegendValue.Value].includes(allowedLegendStats[0].value);

  const isCurrentAndLastValueOnlySelected =
    isSingleStatsOnly && legendStats[0] === LegendValue.CurrentAndLastValue;

  const canShowLegendTitleChangeConfig =
    canShowStatisticsConfigs &&
    !isCurrentAndLastValueOnlySelected &&
    legendStats.length > 0 &&
    !!onLegendTitleChange;
```